rts/motoko-rts/src/types.rs

```diff
@@ -921,6 +921,10 @@ pub(crate) unsafe fn block_size(address: usize) -> Words<u32> {
 
         TAG_BIGINT => {
             let bigint = address as *mut BigInt;
+            if (*bigint).mp_int.used != (*bigint).mp_int.alloc {
+                println!(100, "used:{}, alloc: {}", (*bigint).mp_int.used, (*bigint).mp_int.alloc);
```

Comment:
Building up the `arr` (wins back 1-2 words)
```
debug.print: used:3, alloc: 5
debug.print: used:5, alloc: 7
debug.print: used:9, alloc: 11
debug.print: used:18, alloc: 19
debug.print: used:36, alloc: 37
debug.print: used:72, alloc: 73
debug.print: used:143, alloc: 145
debug.print: used:286, alloc: 287
debug.print: used:572, alloc: 573
```
Deserialising from Candid blob (wins back 1 word consistently)
```
debug.print: used:3, alloc: 4
debug.print: used:5, alloc: 6
debug.print: used:9, alloc: 10
debug.print: used:18, alloc: 19
debug.print: used:36, alloc: 37
debug.print: used:72, alloc: 73
debug.print: used:143, alloc: 144
debug.print: used:286, alloc: 287
debug.print: used:572, alloc: 573
```